### DISCUSSION TITLE
How can I validate filename of the file?

### DISCUSSION THREAD

**Discussion by positivedeveloper36** [adonisjs/core#3846] [Help]
I try like this :

```
const coverImage = request.file('cover_image', {
  size: '2mb',
  extnames: ['jpg', 'png', 'gif'],
  clientName: schema.string([
    rules.regex(new RegExp('[a-zA-Z\d]+_[0-9\d]'))
  ]),
})
```

But it doesn't work. Seems validation file only reads size and extnames

How can I solve this problem

Please help me. Thanks

**Comment by Barbapapazes** [comment]
Hello,

There is [no rules available for 'file'](https://docs.adonisjs.com/reference/validator/schema/file#define-additional-rules)

You must write you own on top of the one for the string validator.

**Reply by positivedeveloper36** [reply]
@Barbapapazes I'm still confused. You mean like this : https://docs.adonisjs.com/guides/validator/custom-rules?

**Reply by Barbapapazes** [reply]
Yes exactly!
